scripts/deployTokenSale.ts

```diff
@@ -0,0 +1,126 @@
+  if (!tokenLockAddress) throw new Error(`Deployment file did not include tokenLock address '${deploymentFilePath}'.`);
+
+  return {
+    governor: ArenaGovernor__factory.connect(governorAddress, signer),
+    arenaToken: ArenaToken__factory.connect(arenaAddress, signer),
+    timelock: TimelockController__factory.connect(timelockAddress, signer),
+    tokenLock: TokenLock__factory.connect(tokenLockAddress, signer),
+  };
+};
+
+task('deployTokenSale', 'deploy token sale and make proposal for relevant actions').setAction(async (taskArgs, hre) => {
+  const networkId = hre.network.config.chainId as number;
+  const [proposer] = await hre.ethers.getSigners();
+  proposerAddress = await proposer.getAddress();
+  console.log(`Proposer: ${proposerAddress}`);
+
+  let config = tokenSaleConfigs[networkId];
+  if (!config) throw new Error(`Unknown network ${hre.network.name} (${networkId})`);
+  const {governor, arenaToken, timelock, tokenLock} = getContracts(proposer, allConfigs[networkId]);
+
+  console.log(`deploying tokensale...`);
+  const TokenSaleFactory = (await hre.ethers.getContractFactory('TokenSale')) as TokenSale__factory;
+
+  tokenSale = await TokenSaleFactory.deploy(
+    config.TOKEN_SALE_USDC,
+    arenaToken.address,
+    config.TOKEN_SALE_START,
+    config.TOKEN_SALE_DURATION,
+    config.TOKEN_SALE_ARENA_PRICE,
+    config.TOKEN_SALE_RECIPIENT,
+    tokenLock.address,
+    allConfigs[networkId].VEST_DURATION
+  );
+  await tokenSale.deployed();
+  console.log(`tokenSale address: ${tokenSale.address}`);
+
+  // set up token sale whitelist
+  await tokenSale.changeWhiteList(
+    config.TOKEN_SALE_WHITELIST.map(({buyer}) => buyer),
+    config.TOKEN_SALE_WHITELIST.map(({arenaAmount}) => arenaAmount)
+  );
+  const TOKEN_SALE_SUPPLY = config.TOKEN_SALE_WHITELIST.reduce((sum, el) => sum.add(el.arenaAmount), BN.from(`0`));
+  // transfer token sale admin role to timelock
+  await tokenSale.transferOwnership(timelock.address);
+
+  // 1st action: set token sale in TokenLock
+  // 2nd action: request TOKEN_SALE_SUPPLY tokens from timelock to tokenSale
+  let targets: string[] = [tokenLock.address, arenaToken.address];
+  let values: string[] = ['0', '0'];
+  let calldatas: string[] = [
+    tokenLock.interface.encodeFunctionData('setTokenSale', [tokenSale.address]),
+    arenaToken.interface.encodeFunctionData('transfer', [tokenSale.address, TOKEN_SALE_SUPPLY]),
+  ];
+
+  const tx = await governor['propose(address[],uint256[],bytes[],string)'](
+    targets,
+    values,
+    calldatas,
+    `Conduct Arena token sale!`
+  );
+  console.log(`proposal submitted: ${tx.hash}`);
+  console.log(`waiting for block inclusion ...`);
+  await tx.wait(1);
+
+  console.log('exporting addresses...');
+  let addressesToExport = {
+    proposer: proposerAddress,
+    tokenSale: tokenSale.address,
+  };
+  let exportJson = JSON.stringify(addressesToExport, null, 2);
+  fs.writeFileSync(config.EXPORT_FILENAME, exportJson);
```

Comment:
do we just want to merge the `tokenSale` with the other deployments in the existing `deployments/${config.EXPORT_FILENAME}` and write it? I don't think the proposer is important